src/status_im/contexts/chat/messenger/photo_selector/effects.cljs

```diff
@@ -29,27 +30,30 @@
         #(rf/dispatch [:on-camera-roll-get-photos (:edges %) (:page_info %) end-cursor])))})))
 
 (defn- resize-photo
-  [uri callback]
-  (rn/image-get-size
-   uri
-   (fn [width height]
-     (let [resize? (> (max width height) maximum-image-size-px)]
-       (image-resizer/resize
-        uri
-        (if resize? maximum-image-size-px width)
-        (if resize? maximum-image-size-px height)
-        60
-        (fn [^js resized-image]
-          (let [path (.-path resized-image)
-                path (if (string/starts-with? path "file") path (str "file://" path))]
-            (callback {:resized-uri path
-                       :width       width
-                       :height      height})))
-        #(log/error "could not resize image" %))))))
+  [uri]
+  (p/let [[width height] (rn/image-get-size uri)
+          resize?        (> (max width height)
+                            maximum-image-size-px)
+          resized-uri    (-> (image-resizer/resize
+                              {:max-width  (if resize? maximum-image-size-px width)
+                               :max-height (if resize? maximum-image-size-px height)
+                               :path       uri
+                               :quality    60})
+                             (p/then (fn [^js resized-image]
+                                       (let [path (.-path resized-image)]
+                                         (if (string/starts-with? path "file")
+                                           path
+                                           (str "file://" path))))))]
+    {:resized-uri resized-uri
+     :width       width
+     :height      height}))
 
 (rf/reg-fx :effects.camera-roll/image-selected
  (fn [[image chat-id]]
-   (resize-photo (:uri image) #(rf/dispatch [:photo-selector/image-selected chat-id image %]))))
+   (-> (resize-photo (:uri image))
+       (p/then #(rf/dispatch
+                 [:photo-selector/image-selected chat-id image %]))
+       (p/catch #(log/error "could not resize image" %)))))
```

Comment:
IMO, we should use promesa as a tool to handle existing promises from the JS world, but here we are switching the approach from callbacks to promises, and honestly I don't think it's cleaner/simpler.

Let's take this example to use `resize-photo`.
Instead of just  passing a Clojure function that will be called (it's simple), we must deal with js object invocations (`.then`), or always require promesa (` [promesa.core :as p]`) and then use its functions/macros (`p/then`, `p/catch`) to just call the function we wanted in the first place. It looks like boilerplate code for me.

On the other hand, I see the value of Promesa to work with promises coming from the JS world, like this:
[RN async storage](https://react-native-async-storage.github.io/async-storage/docs/api):
![image](https://github.com/status-im/status-mobile/assets/90291778/5b919f5a-7093-48b7-bee2-90f19c57032c)

For that `await` instead of using `.then` we use `p/let`

Reply:
@ulisesmac I see your point. I was debating whether it's useful or not to convert it to a promise, but decided to add it and see what the feedback is 😄. In case you disagree with the below, will remove it.

IMO using .then and .catch is a small price to pay for _composition_, _consistency_ and _error handling_, but I may be biased. As a promise it can be: 
- composed with other promises (as shown above);
- consistend in how we deal with asynchronicity (in this case in the same function we'd have a callback with promises resolved inside). Not awful, but mixing them feels odd;
- on failure the promise will be rejected and handled down the line. Otherwise, the `on-fail` feels optional and most often will just not be used, as can be seen in `react-native/resize-photo` where it's completely ignored;

Also, the consumer of `resize-photo` doesn't have to use `promesa` necessarily and the only downside seems to be the addition of `.then`/`.catch`, which is "micro"-boilerplate to me 😄.

Reply:
hey @clauxx 
I understand your perspective, but I might be biased too.

> consistend in how we deal with asynchronicity (in this case in the same function we'd have a callback with promises resolved inside). Not awful, but mixing them feels odd;

I see it more as a simplification, the consumer doesn't need to deal with JS Object calls, I feel it more in the functional programming way.

> I was debating whether it's useful or not to convert it to a promise, but decided to add it and see what the feedback is 

Yeah, that's good :+1:

 This will affect the way we write async code, so let's see what is the perspective of other team members.

Reply:
Well @clauxx 
It seems I'm the only one thinking like this, idk.

So let's merge it then :+1: